Library/Source/NativeEngine.cpp

```diff
@@ -1081,6 +1198,71 @@ namespace babylon
         delete textureData;
     }
 
+    Napi::Value NativeEngine::Impl::CreateFrameBuffer(const Napi::CallbackInfo& info)
+    {
+        const auto textureData = info[0].As<Napi::External<TextureData>>().Data();
+        uint16_t width = static_cast<uint16_t>(info[1].As<Napi::Number>().Uint32Value());
+        uint16_t height = static_cast<uint16_t>(info[2].As<Napi::Number>().Uint32Value());
+        bgfx::TextureFormat::Enum format = static_cast<bgfx::TextureFormat::Enum>(info[3].As<Napi::Number>().Uint32Value());
+        int samplingMode = info[4].As<Napi::Number>().Uint32Value();
+        bool generateStencilBuffer = info[5].As<Napi::Boolean>();
+        bool generateDepth = info[6].As<Napi::Boolean>();
+        bool generateMipMaps = info[7].As<Napi::Boolean>();
+
+        bgfx::FrameBufferHandle frameBufferHandle{};
+        if (generateStencilBuffer && !generateDepth)
+        {
+            throw std::exception{ /* Does this case even make any sense? */ };
+        }
+        else if (!generateStencilBuffer && !generateDepth)
+        {
+            frameBufferHandle = bgfx::createFrameBuffer(width, height, TEXTURE_FORMAT[format], BGFX_TEXTURE_RT);
+        }
+        else
+        {
+            auto depthStencilFormat = bgfx::TextureFormat::D32;
+            if (generateStencilBuffer)
+            {
+                depthStencilFormat = bgfx::TextureFormat::D24S8;
+            }
+
+            std::array<bgfx::TextureHandle, 3> textures
+            {
+                bgfx::createTexture2D(width, height, generateMipMaps, 1, TEXTURE_FORMAT[format], BGFX_TEXTURE_RT),
+                bgfx::createTexture2D(width, height, generateMipMaps, 1, TEXTURE_FORMAT[format], BGFX_TEXTURE_RT),
+                bgfx::createTexture2D(width, height, generateMipMaps, 1, depthStencilFormat, BGFX_TEXTURE_RT)
+            };
+            std::array<bgfx::Attachment, 3> attachments{};
+            for (int idx = 0; idx < attachments.size(); ++idx)
+            {
+                attachments[idx].init(textures[idx]);
+            }
+            frameBufferHandle = bgfx::createFrameBuffer(attachments.size(), attachments.data(), true);
+        }
+
+        textureData->Texture = bgfx::getTexture(frameBufferHandle);
+
+        return Napi::External<FrameBufferData>::New(info.Env(), new FrameBufferData(frameBufferHandle, width, height));
+    }
+
+    void NativeEngine::Impl::BindFrameBuffer(const Napi::CallbackInfo& info)
+    {
+        const auto frameBufferData = info[0].As<Napi::External<FrameBufferData>>().Data();
+        m_boundFrameBuffer = frameBufferData;
+
+        // TODO: Consider doing this only on bgfx::reset(); the effects of this call don't survive reset, but as
+        // long as there's no reset this doesn't technically need to be called every time the frame buffer is bound.
+        m_boundFrameBuffer->SetUpView();
+
+        // bgfx::setTexture()? Why?
+        // TODO: View order?
```

Comment:
Added lines for both TODOs in https://github.com/BabylonJS/BabylonNative/issues/6.